### PR TITLE
*: merge ranger.IndexRange and ranger.ColumnRange.

### PR DESCRIPTION
**Pull request by winoros**
Currently `IntColumnRange` doesn't support calculate range if it exceed int64. Modify `IntColumnRange` to fit this is no better than use `ColumnRange` directly.
And we can merge `ColumnRange` and `IndexRange` at the same time. So there's only one type of range。
PTAL @coocood @hanfei1991 @lamxTyler

### CODE REVIEW

**Comment by coocood** [comment]
Why make this change?

**Comment by winoros** [comment]
@coocood updated

**Comment by shenli** [comment]
@hanfei1991 @lamxTyler PTAL

File: executor/executor_pkg_test.go [hunk 1]
```diff
@@ -179,11 +179,11 @@ func (s *testExecSuite) TestBuildKvRangesForIndexJoin(c *C) {
 	}
 }
 
-func generateIndexRange(vals ...int64) *ranger.IndexRange {
+func generateIndexRange(vals ...int64) *ranger.NewRange {
```
Comment (lamxTyler) [inline executor/executor_pkg_test.go]: should be `generateNewRange`.

File: statistics/histogram.go [hunk 2]
```diff
@@ -512,18 +512,18 @@ func (c *Column) getIntColumnRowCount(sc *stmtctx.StatementContext, intRanges []
 }
 
 // getColumnRowCount estimates the row count by a slice of ColumnRange.
```
Comment (lamxTyler) [inline statistics/histogram.go]: Update the comment.

File: statistics/table.go [hunk 3]
```diff
@@ -311,16 +311,16 @@ func (t *Table) GetRowCountByIntColumnRanges(sc *stmtctx.StatementContext, colID
 }
 
 // GetRowCountByColumnRanges estimates the row count by a slice of ColumnRange.
-func (t *Table) GetRowCountByColumnRanges(sc *stmtctx.StatementContext, colID int64, colRanges []*ranger.ColumnRange) (float64, error) {
+func (t *Table) GetRowCountByColumnRanges(sc *stmtctx.StatementContext, colID int64, colRanges []*ranger.NewRange) (float64, error) {
```
Comment (lamxTyler) [inline statistics/table.go]: Update the comment.

File: util/ranger/ranger.go [hunk 4]
```diff
@@ -383,7 +353,7 @@ func fixRangeDatum(v *types.Datum, length int) {
 	}
 }
 
-// BuildRange is a method which can calculate IntColumnRange, ColumnRange, IndexRange.
+// BuildRange is a method which can calculate IntColumnRange, ColumnRange, NewRange.
```
Comment (lamxTyler) [inline util/ranger/ranger.go]: Update the comment.

File: util/ranger/types.go [hunk 5]
```diff
@@ -63,88 +62,45 @@ func (tr IntColumnRange) Convert2IntRange() IntColumnRange {
...
-}
-
-// Convert2IndexRange implements the Convert2IndexRange interface.
-func (cr *ColumnRange) Convert2IndexRange() *IndexRange {
-	panic("you shouldn't call this method.")
-}
-
-// IndexRange represents a range for an index.
-type IndexRange struct {
+// NewRange represents a range for an index.
```
Comment (lamxTyler) [inline util/ranger/types.go]: Not only index?

File: util/ranger/types.go [hunk 6]
```diff
@@ -63,88 +62,45 @@ func (tr IntColumnRange) Convert2IntRange() IntColumnRange {
...
 	}
-	for i, length := 0, len(ir.HighVal); i < length; i++ {
-		newRange.HighVal = append(newRange.HighVal, ir.HighVal[i])
+	for i, length := 0, len(ran.HighVal); i < length; i++ {
+		newRange.HighVal = append(newRange.HighVal, ran.HighVal[i])
 	}
 	return newRange
 }
 
 // IsPoint returns if the index range is a point.
```
Comment (lamxTyler) [inline util/ranger/types.go]: Update the comment

File: executor/executor_pkg_test.go (hunk 1 above)
Comment (winoros) [reply executor/executor_pkg_test.go]: There is specified as index's range.

**Comment by coocood** [comment]
LGTM

**Pull request merged**